### ISSUE TITLE
Any issues with HipChat rate limiting?

### ISSUE THREAD

**Issue by amorphic** [skoczen/will#140]
Hello,

I've been experimenting with both Err and Will connected to our instance of HipChat Server.

I notice that the maintainers of Err were previously using HipChat's API for their HipChat backend but they ultimately abandoned it for a number of reasons, chiefly HipChat's API rate limiting: https://github.com/gbin/err/commit/aef572ab790e0510d247499087039b8f8bf185fd.

Sadly because HipChat [does not support XHTML-IM|http://help.hipchat.com/forums/138883-suggestions-ideas/suggestions/2974079-support-xhtml-im-send-text-only-notifications-to] this means that Err is unable to publish rich-content messages to HipChat, which is a bit of a shame.

Have you come across any similar issues with Will? I would assume that if you're using the same API you would encountered the same problems, but I haven't been able to to find anybody reporting rate limiting as an issue.

Apologies for the slightly fluffy Issue...

**Comment by skoczen** [comment]
Hey @amorphic ,

Thanks for your question!   I'm putting all that I know below - if you've got suggestions on a better place where this stuff could live in the docs, I'm all ears.

I can't say we have - I've not heard of anyone running into rate-limit problems.  We're using XMPP to send the messages though, not the API, so the limit is different. I've been using two instances of will (one work, one personal) for a year and a half, without trouble.  The limit is ~150 messages / minute (this will go up to ~250 when some work to use the new tokens is done.)  It's conceivable that one could hit that, but it's a *lot* of talking for a bot to do.  

The HTML stuff is a bit of a pain, but most of us have worked around it by doing a plain-text message with mentions right after the HTML one.   I'm sure you read through that err issue, but to be clear, HTML messages work fine, they're just not *also* auto-highlighted (with mentions, github and image parsing, etc.)  Personally, I look at that as a feature, not a bug - when I've got content that I took the time to format HTML for, I generally don't want hipchat auto-converting it.  Mentions are the lone caveat, but for every purpose I've run across, a quick normal-message mention does trick well.

Let me know if you've got any other questions!

-Steven

**Comment by amorphic** [comment]
Hey Steven, thanks for getting back to me so quickly.

I didn't realise that Will was using XMPP for messaging and also that HipChat XMPP was rate-limited. We're trialling HipChat Server but I assume the limits are the same as for hipchat.com.

Thanks for your suggestion re posting html vs plain text!

**Comment by skoczen** [comment]
@amorphic - absolutely!  Quick replies (even if they aren't fixes) is part of [what we're trying to do](http://skoczen.github.io/will/improve/) to keep an active, engaged community.  

Backend - it's actually a mix - XMPP isn't limited (as far as I know), but it's only used for message sending.  Everything else (get member info, list of rooms, other miscellany) is done via the API.  It's even more complex under the hood, between the V1 and V2 api, but:

TL;DR: Neither I nor any of the lots of people who are using will have run into any rate-limit problems.  It's no guarantee, but there are enough people using will (including those who are using HipChat Server), that I'd assume I would have heard if we had troubles on that front!

Please open up an issue if you run into any snags!
